pkg/jx/cmd/upgrade_cluster.go

```diff
@@ -63,6 +63,24 @@ func NewCmdUpgradeCluster(f cmdutil.Factory, out io.Writer, errOut io.Writer) *c
 
 // Run implements the command
 func (o *UpgradeClusterOptions) Run() error {
+	confirm := false
```

Comment:
LGTM! We could use `if !o.BatchMode` to avoid prompting for confirmation if folks use batch mode (`-b`) - not all commands have a batch mode but its handy if folks wanna use `jx` inside shells scripts etc. Maybe we could add a `--yes` CLI argument also to avoid the confirm?